### PR TITLE
Flatpak

### PR DESCRIPTION
**Pull request by mattdangerw**
https://phabricator.endlessm.com/T13552

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,9 +208,9 @@ Function that renames all files generated by the task just before putting them i
 Type: `String`
 Default: `package.name`
 
-Name of the package (e.g. `atom`), used in the [`Package` field of the `control` specification](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Package).
+Name of the package (e.g. `atom`), used in the [`Package` field of the `control` specification](https://www.flatpak.org/doc/flatpak-policy/ch-controlfields.html#s-f-Package).
```
Comment (ptomato) [inline README.md]: URLs probably are not right, same goes for Flatpak Policy Manual below :smile:

File: README.md [hunk 2]
```diff
@@ -380,20 +370,21 @@ MIME types the application is able to open, used in the [`MimeType` field of the
...
 
 ## Contributors
 
 * Daniel Perez Alvarez ([unindented@gmail.com](mailto:unindented@gmail.com))
+* Matt Watson ([mattdangerw@gmail.com](mailto:mattdangerw@gmail.com))
 
 
 ## License
 
-Copyright (c) 2016 Daniel Perez Alvarez ([unindented.org](https://unindented.org/)). This is free software, and may be redistributed under the terms specified in the LICENSE file.
```
Comment (ptomato) [inline README.md]: Original author still retains copyright.

File: src/installer.js [hunk 3]
```diff
@@ -185,22 +185,10 @@ var createDesktop = function (options, dir, callback) {
 }
 
 /**
- * Create pixmap icon for the package.
- */
-var createPixmapIcon = function (options, dir, callback) {
-  var iconFile = path.join(dir, 'share/pixmaps', options.id + '.png')
```
Comment (ptomato) [inline src/installer.js]: AFAIK `share/pixmaps` is like 15 years old :confused:

File: src/installer.js [hunk 4]
```diff
@@ -215,9 +203,10 @@ var createHicolorIcon = function (options, dir, callback) {
  */
 var createIcon = function (options, dir, callback) {
   if (_.isObject(options.icon)) {
-    createHicolorIcon(options, dir, callback)
+    createHicolorIcon(options.icon, options, dir, callback)
   } else {
-    createPixmapIcon(options, dir, callback)
+    // FIXME: hacky!! Should we read the resolution ourselves??
```
Comment (ptomato) [inline src/installer.js]: Yep, I think only SVGs are supposed to go in `scalable`

File: .travis.yml [hunk 5]
```diff
@@ -13,11 +13,10 @@ addons:
...
   provider: npm
-  email: unindented@gmail.com
   api_key:
     secure: gA6QGEyXv6rOHHv9OKurFaPlhDjIbyxGF+45vamtkPlL/82pDErd/GK/oWWEP9MYdyjcp4QpRlLn/CTXSaFB6RaA8ceR2PwS7J02f5jZPOMDGOUD/apyoUKSaIEG1tVDegaoNCuMEKSG7p8zGuZEJf45Qwzfkm1rkQ7AicG9wzNrv5IMZzETJcdSAJyaVV3jm3FP+o6QdsOg1M9I+rqJaVsTc4tO5t+sUCMRCtZkrBmRQ+p2jbCl+gYXPSMCbw3dMso/gdqEcHCs1LWi5oj5Jlo94zP6i0zEEdZMagj7IDwvVAgbA/OTjhcEsKrNj4nJEa0JNLjvYdq2LZ6cQWFdWR8OVnYHbhqG9dXEeuVDkwQcnlp3+7FioZ7DCzybck6XZhBepXYY6h3J08jCf9lpiQBs/U/pij20f39wx1tTM3hzlfespX0qpejukak7p47ko050udNH7B1C64NvFi7q2Vg2h8jEZ4rYZ1qYmzcopTV54zhRzoyKIuH3+Z5x2cb85NSW5KSZnYXDf2o/SDColDz9se8uunf0PlkY2fZP+1t42QbyVI6JGZICiXdoX1CEUneGV+iiPDCEGVTm1LgaN6cORuLZz4LKIerxV6NRNIYODwaJVQ3q24vwndVeXqrNic1wJRw0B9F5t3ttRcPqsvW+XRa1wVxEqQUaNp9BoMk=
   on:
     tags: true
 
 after_success:
-  - lintian test/fixtures/out/*.deb
+  - lintian test/fixtures/out/*.flatpak
```
Comment (ptomato) [inline .travis.yml]: Probably want to remove this `lintian` line

File: .travis.yml [hunk 6]
```diff
@@ -13,11 +13,10 @@ addons:
 
 deploy:
   provider: npm
-  email: unindented@gmail.com
   api_key:
     secure: gA6QGEyXv6rOHHv9OKurFaPlhDjIbyxGF+45vamtkPlL/82pDErd/GK/oWWEP9MYdyjcp4QpRlLn/CTXSaFB6RaA8ceR2PwS7J02f5jZPOMDGOUD/apyoUKSaIEG1tVDegaoNCuMEKSG7p8zGuZEJf45Qwzfkm1rkQ7AicG9wzNrv5IMZzETJcdSAJyaVV3jm3FP+o6QdsOg1M9I+rqJaVsTc4tO5t+sUCMRCtZkrBmRQ+p2jbCl+gYXPSMCbw3dMso/gdqEcHCs1LWi5oj5Jlo94zP6i0zEEdZMagj7IDwvVAgbA/OTjhcEsKrNj4nJEa0JNLjvYdq2LZ6cQWFdWR8OVnYHbhqG9dXEeuVDkwQcnlp3+7FioZ7DCzybck6XZhBepXYY6h3J08jCf9lpiQBs/U/pij20f39wx1tTM3hzlfespX0qpejukak7p47ko050udNH7B1C64NvFi7q2Vg2h8jEZ4rYZ1qYmzcopTV54zhRzoyKIuH3+Z5x2cb85NSW5KSZnYXDf2o/SDColDz9se8uunf0PlkY2fZP+1t42QbyVI6JGZICiXdoX1CEUneGV+iiPDCEGVTm1LgaN6cORuLZz4LKIerxV6NRNIYODwaJVQ3q24vwndVeXqrNic1wJRw0B9F5t3ttRcPqsvW+XRa1wVxEqQUaNp9BoMk=
```
Comment (ptomato) [inline .travis.yml]: This is probably not the right API key

Although I didn't know that Travis CI could deploy stuff! That is cool!

File: README.md [hunk 7]
```diff
@@ -1,33 +1,25 @@
...
 ## Requirements
 
-This tool requires `fakeroot` and `dpkg` to build the `.deb` package.
-
-I'd recommend building your packages on your target platform, but if you insist on using Mac OS X, you can install these tools through [Homebrew](http://brew.sh/):
-
-```
-$ brew install fakeroot dpkg
-```
+This tool requires `flatpak` to be actually installed on your system.
```
Comment (ptomato) [inline README.md]: Here too, should mention v0.6.13.

File: README.md [hunk 8]
```diff
@@ -238,73 +223,82 @@ Generic name of the application (e.g. `Text Editor`), used in the [`GenericName`
...
-Application area into which the package has been classified, used in the [`Section` field of the `control` specification](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Section).
+Base app version, used in the [`base-version` field of a flatpak-builder manifest](http://flatpak.org/flatpak/flatpak-docs.html#flatpak-builder).
 
-You can read more about [sections](https://www.debian.org/doc/debian-policy/ch-archive.html#s-subsections), and also check out the [list of existing sections in Debian unstable](https://packages.debian.org/unstable/).
 
-#### options.priority
+#### options.baseFlatpakref
 Type: `String`
-Default: `"optional"`
+Default: `FIXME`
```
Comment (ptomato) [inline README.md]: What goes here?

**Review by ptomato — COMMENTED** [review]
Seems reasonable, just a few things I noticed in the documentation.

File: README.md (hunk 8 above)
Comment (mattdangerw) [reply README.md]: Same as baseFlatpakref in `flatpak-bundler`, a url or local path to a flatpakref file. So we can autoinstall the electron base app.

File: .travis.yml (hunk 6 above)
Comment (mattdangerw) [reply .travis.yml]: I'm gonna hold off fixing testing to https://phabricator.endlessm.com/T13989

**Pull request closed without merging**